python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -131,14 +131,31 @@ def dist_inference_one_layer(layer_id, g, dataloader, target_ntypes, layer, get_
     -------
     dict of Tensors : the inferenced tensors.
     """
+    len_dataloader = len(list(dataloader))
+    barrier()
+    tensor = th.tensor([len_dataloader], device=device)
+    th.distributed.all_reduce(tensor, op=th.distributed.ReduceOp.MAX)
+    max_num_batch = tensor[0]
+
+    dataloader_iter = iter(dataloader)
     y = {}
-    for iter_l, (input_nodes, output_nodes, blocks) in enumerate(dataloader):
+
+    # To use WholeGraph for feature featching, Dataloaders from different
+    # trainers must iterate through the same number of iterations as WholeGraph
+    # does not support imbalanced batch numbers across processes/trainers
```

Comment:
this implementation is quite hacky. can you add a TODO here? we need to fix it afterwards.

Reply:
Added